### DISCUSSION TITLE
'Local Docs' option

### DISCUSSION THREAD

**Discussion by toafloast** [godotengine/godot-proposals#7467] [Editor]
I often find myself without internet, and I think it would be nice to have a 'Local Docs' option. I downloaded the docs locally, so I can access them offline- but I think it would be nice if the Editor could have the option to access the local docs when clicking 'Online Docs' 
(replaced by Local Docs- either when no wifi is detected or manually by user choice). The path to the root of the local docs could be set in Project or Editor settings, whichever makes more sense. 
![image](https://github.com/godotengine/godot-proposals/assets/104696964/ecb67f0b-c25f-437d-a05d-f939d675b964)
Why should it be added? Well, it's a nice QOL improvement, for one. It could also possibly be expanded upon such that if a local docs path is set, it can be browsed in-editor without having to leave it.

**Comment by toafloast** [comment]
![image](https://github.com/godotengine/godot-proposals/assets/104696964/3dba62c6-56ac-407c-8117-2bd10d41044f)
Heres the folder it would be given access to. Just the html docs download from github.

**Comment by Calinou** [comment]
It shouldn't be too difficult to implement this, but I'm concerned about discoverability of this feature.

One way we could improve discoverability is by performing an HTTP request against `docs.godotengine.org` before opening the browser. If the request fails, we assume online documentation is unavailable and show a dialog asking the user to point to a local documentation path.

**Reply by theraot** [reply]
Could Godot handle local documentation similar to export templates?

**Reply by Calinou** [reply]
> Could Godot handle local documentation similar to export templates?

Probably, but this would increase implementation complexity as we'd need to implement a downloading/installing mechanism.

**Reply by toafloast** [reply]
I am in favor of the simplest approach of setting a local path, and enabling it through settings, as it is a rather niche use. There is another thing to consider, as to how the various tutorial links of the in-editor docs will be handled, iirc some link to external sites like youtube. will those be disabled? will other links open local files? etc

**Reply by Calinou** [reply]
> There is another thing to consider, as to how the various tutorial links of the in-editor docs will be handled, iirc some link to external sites like youtube. will those be disabled? will other links open local files? etc

Editor help uses a `$DOCS_URL` placeholder, which could be replaced with a local link if the setting is set to a valid path for the documentation. External links will be attempted as-is though.
